### DISCUSSION TITLE
SetUserMatrix of PolyData not working in Trame Client Mode

### DISCUSSION THREAD

**Discussion by m-vik** [pyvista/pyvista#4332] [Q&A]
Hello,
I faced a problem between server and client mode of trame backend. I am generating a surface with the vtkRotationalExtrusionFilter and creating an actor. The actor is then transformed in each time step by using SetUserMatrix.

This works fine in trame server mode, but in client mode the SetUserMatrix input is ignored and the generated surface stays at global origin.

I managed to get it kind of working with vtkTransformPolyDataFilter, but it slowing down the plot and jitter/jumps (not smooth).

Is there a reason why SetUserMatrix is working when rendered on server side and not on client side?

**Comment by banesullivan** [comment]
This seems like a missing feature in the `vtkActor` serializer in `trame-vtk`: https://github.com/Kitware/trame-vtk

See https://github.com/Kitware/trame-vtk/blob/9509c69ddac981f7f5c30e1da0f741a2d32497ab/trame_vtk/modules/vtk/serializers/actors.py#L301 for the implementation for axes actors. It looks like this uses the `GetUserTransform()` instead of `GetUserMatrix()`. I wonder if we can support both in the implementation for generic actors?

cc @jourdain, @psavery

**Comment by jourdain** [comment]
I think we can. We did it for the orientation actor (to some extent) and it should be ported to the general actor serializer.

**Comment by m-vik** [comment]
Thank you for the very fast response. None of both is working on client side:
```python
transform = vtk.vtkTransform(); transform.SetMatrix(T_perp)
actor.SetUserTransform(transform)
# or
actor.SetUserMatrix(T_perp)
```

Or is this basically the same under the hood?

**Reply by jourdain** [reply]
(previous comment) trame-vtk needs a small fix to support it.
